radiation/python/radsw/radsw_main.py

```diff
@@ -0,0 +1,4031 @@
+
+                    for ib in range(nbdsw):
+                        flxu0[k] = flxu0[k] + fxup0[k, ib]
+                        flxd0[k] = flxd0[k] + fxdn0[k, ib]
+
+            #  --- ...  prepare for final outputs
+            for k in range(nlay):
+                rfdelp[k] = self.heatfac / delp[k]
+
+            if self.lfdncmp:
+                # --- ...  optional uv-b surface downward flux
+                uvbf0[j1] = suvbf0
+                uvbfc[j1] = suvbfc
+
+                # --- ...  optional beam and diffuse sfc fluxes
+                nirbm[j1] = sfbmc[0]
+                nirdf[j1] = sfdfc[0]
+                visbm[j1] = sfbmc[1]
+                visdf[j1] = sfdfc[1]
+
+            #  --- ...  toa and sfc fluxes
+
+            upfxc_t[j1] = ftoauc
+            dnfxc_t[j1] = ftoadc
+            upfx0_t[j1] = ftoau0
+
+            upfxc_s[j1] = fsfcuc
+            dnfxc_s[j1] = fsfcdc
+            upfx0_s[j1] = fsfcu0
+            dnfx0_s[j1] = fsfcd0
+
+            #  --- ...  compute heating rates
+
+            fnet[0] = flxdc[0] - flxuc[0]
+
+            for k in range(1, nlp1):
+                fnet[k] = flxdc[k] - flxuc[k]
+                hswc[j1, k - 1] = (fnet[k] - fnet[k - 1]) * rfdelp[k - 1]
+
+            # --- ...  optional flux profiles
+
+            if self.lflxprf:
+                for k in range(nlp1):
+                    upfxc_f[j1, k] = flxuc[k]
+                    dnfxc_f[j1, k] = flxdc[k]
+                    upfx0_f[j1, k] = flxu0[k]
+                    dnfx0_f[j1, k] = flxd0[k]
+
+            # --- ...  optional clear sky heating rates
+
+            if self.lhsw0:
+                fnet[0] = flxd0[0] - flxu0[0]
+
+                for k in range(1, nlp1):
+                    fnet[k] = flxd0[k] - flxu0[k]
+                    hsw0[j1, k - 1] = (fnet[k] - fnet[k - 1]) * rfdelp[k - 1]
+
+            # --- ...  optional spectral band heating rates
+
+            if self.lhswb:
+                for mb in range(nbdsw):
+                    fnet[0] = fxdnc[0, mb] - fxupc[0, mb]
+
+                    for k in range(nlay):
+                        fnet[k + 1] = fxdnc[k + 1, mb] - fxupc[k + 1, mb]
+                        hswb[j1, k, mb] = (fnet[k + 1] - fnet[k]) * rfdelp[k]
+
+        return (
+            hswc,
+            upfxc_t,
+            dnfxc_t,
+            upfx0_t,
+            upfxc_s,
+            dnfxc_s,
+            upfx0_s,
+            dnfx0_s,
+            cldtau,
+            hsw0,
+            uvbf0,
+            uvbfc,
+            nirbm,
+            nirdf,
+            visbm,
+            visdf,
+        )
+
+    def cldprop(
+        self,
+        cfrac,
+        cliqp,
+        reliq,
+        cicep,
+        reice,
+        cdat1,
+        cdat2,
+        cdat3,
+        cdat4,
+        cf1,
+        nlay,
+        ipseed,
+        dz,
+        delgth,
+        ipt,
+    ):
+        ds = xr.open_dataset("../lookupdata/radsw_cldprtb_data.nc")
+        extliq1 = ds["extliq1"].data
```

Comment:
why do you need the ".data"? oh these are probably storages and for the python version you need the numpy array?

Reply:
yes, especially when I move to gt4py, since I need to add dimensions etc